gateway/api.go

```diff
@@ -681,11 +681,14 @@ func (gw *Gateway) handleAddKey(keyName, hashedName, sessionString, apiID string
 		return
 	}
 
+	lifetime := GetSessionLifetime(sess, gw)
+
 	var err error
 	if gw.GetConfig().HashKeys {
```

Comment:
Maybe it could be cleaner if we create a key var for hashed/unhashed and we just call `gw.GlobalSessionManager.UpdateSession(keyVAR, sess, lifetime, gw.GetConfig().HashKeys)`?